src/blr/cli/buildmolecules.py

```diff
@@ -74,7 +76,7 @@ def build_molecules(pysam_openfile, barcode_tag, window, min_reads, summary):
             continue
 
         # Fetches barcode and genomic position. Position will be formatted so start < stop.
-        barcode = fetch_bc(pysam_read=read, barcode_tag=barcode_tag, summary=summary)
+        barcode = utils.get_bamtag(pysam_read=read, tag=barcode_tag)
         if barcode and not read.is_unmapped:
             read_start, read_stop = sorted((read.reference_start, read.reference_end))
 
```

Comment:
```suggestion
elif not barcode:
   summary.reads_without_barcode += 1
```

Since this was removed in the local function we can add this to continue to track reads without barcodes.

Reply:
I also moved the `read.is_unmapped` check to the top so the main loop could be put in the `else` clause instead.